src/agent/centralized_configuration/src/centralized_configuration.cpp

```diff
@@ -10,32 +13,50 @@ namespace centralized_configuration
     {
         try
         {
+            std::vector<std::string> groupIds {};
+            std::string messageOnSuccess {};
 
             if (command == "set-group")
             {
                 if (m_setGroupIdFunction && m_downloadGroupFilesFunction)
                 {
                     if (parameters.empty())
+                    {
+                        LogWarn("Group set failed, no group list");
                         co_return module_command::CommandExecutionResult {
                             module_command::Status::FAILURE,
                             "CentralizedConfiguration group set failed, no group list"};
+                    }
 
-                    const auto groupIds = parameters[0].get<std::vector<std::string>>();
+                    groupIds = parameters[0].get<std::vector<std::string>>();
+                    messageOnSuccess = "CentralizedConfiguration group set";
 
                     m_setGroupIdFunction(groupIds);
+                    m_saveGroupIdFunction();
```

Comment:
I think we could just save the groups in the same function that was passed as set group id function. Like:

```
m_centralizedConfiguration.SetGroupIdFunction([this](const std::vector<std::string>& groups)                                                 {
    m_agentInfo.SetGroups(groups);
    m_agentInfo.SaveGroups();
    return true;
});
```

I think that you can do some error checking instead of just returning `true`, but that's the general idea.

Reply:
Done!